connectors/cli.py

```diff
@@ -87,6 +88,18 @@ def _parser():
     return parser
 
 
+async def _start_service(config, args, loop):
+    preflight = PreflightCheck(config)
+    for sig in (signal.SIGINT, signal.SIGTERM):
+        loop.add_signal_handler(sig, functools.partial(preflight.shutdown, sig))
```

Comment:
I think your hang issue is here; you are registering several functions against SIGTERM. You should only have one that iterates on all services:

```
def _shutdown_service(sig):
    for service in running_services:
        service.shutdown()
```